### DISCUSSION TITLE
What do you think about transpiling `fut` to C?

### DISCUSSION THREAD

**Discussion by ghost** [pfusik/cito#105] [Q&A] [answered]
C++20 is a pretty high requirement that can't be met on some legacy systems stuck with old compilers. This is when C comes to the rescue as you said the generated C code need at most C11 and C11 is widely supported now.

**Comment by pfusik** [comment, marked as answer]
Support for legacy systems is not my priority.
There's some work that would be needed in the C backend, such as runtime type checking (`dynamic_cast` in C++).
